src/sdk/SessionKeyValidator/SessionKeyValidator.ts

```diff
@@ -460,4 +473,14 @@ export class SessionKeyValidator {
             throw new Error(err.message)
         }
     }
+
+    async isAValidToken(token: string): Promise<boolean> {
+
+        const erc20 = new Contract(token, ERC20_ABI, this.provider);
+        const decimals = await erc20.decimals();
+        if (!decimals || decimals == null || decimals as number == 0) {
```

Comment:
@kanthgithub `!decimals` will shadow `decimals==null`
i think we can skip `decimals==null` as it will get covered in `!decimals`

also can a erc20 token not have decimals as zero, is there a reason not to support it ?

Reply:
updated by removing redundant check